src/spring-cloud/azext_spring_cloud/_resource_quantity.py

```diff
@@ -0,0 +1,34 @@
+import re
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+def validate_cpu(cpu):
+    if cpu is None:
+        return None
+
+    if not re.match(r"^\d+m?$", cpu):
+        raise CLIError("CPU quantity should be millis (500m) or integer (1, 2, ...)")
```

Comment:
Could you add some explanation for the function of the regular expression?

Reply:
The current released version only supports integer quantity ([1, 8]). The change being reviewed introduced the notation `###m` for millis, and the only supported fractional amount is `500m`. However, the user is also allowed to specify `1000m` for `1`.

The regex here is to validate the input from the client side, with some digits followed by optional `m`. I added some comments in the updated code.